lib/spack/docs/pipelines.rst

```diff
@@ -59,47 +59,54 @@ Functional Example
 ------------------
 
 The simplest fully functional standalone example of a working pipeline can be
-examined live at this example `project <https://gitlab.com/scott.wittenburg/spack-pipeline-demo>`_
+examined live at this example `project <https://gitlab.com/spack/pipeline-quickstart>`_
 on gitlab.com.
 
 Here's the ``.gitlab-ci.yml`` file from that example that builds and runs the
 pipeline:
 
 .. code-block:: yaml
 
-   stages: [generate, build]
+   stages: [ "generate", "build" ]
 
    variables:
-     SPACK_REPO: https://github.com/scottwittenburg/spack.git
-     SPACK_REF: pipelines-reproducible-builds
+     SPACK_REPOSITORY: "https://github.com/spack/spack.git"
+     SPACK_REF: "develop-2024-10-06"
```

Comment:
In general I don't think it's a good idea to run pipelines with a branch that has the potential to change as frequently as `develop`. In my opinion, the probability of running `generate` with one sha and subsequent build jobs with a different one should be zero.

Reply:
That's fair. My worry is that we are likely going to start dropping snapshot tags before long and only keep the last 6 and this tag won't exist anymore. I don't want to advocate for a hash, I don't think that is better.

Since we haven't started pruning snapshots maybe we can just keep this in mind for things we need to be aware of.

Reply:
I was thinking once we have a release with your mirror credentials change in it, we could switch to that released version.